[Hub Generated] Review request for Microsoft.AgFoodPlatform to add version preview/2021-03-31-preview

RGKarthik opened the pull request:
This is a PR generated at [OpenAPI Hub](https://aka.ms/openapihub). You can view your [work branch via this link](https://portal.azure-devex-tools.com/branch/Azure/azure-rest-api-specs/dev-agfood-Microsoft.AgFoodPlatform-2021-03-31-preview...master/).


### Changelog
Please ensure to add changelog with this PR by answering the following questions.
  1. What's the purpose of the update?    
      - [x] new service onboarding - **AgFood platform. Commercially branded as FarmBeats**
      - [ ] new API version 
      - [ ] update existing version for new feature 
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify 
  2. When you are targeting to deploy new service/feature to public regions? Please provide date, or month to public if date is not available yet. **10th of May**
  3. When you expect to publish swagger? Please provide date, or month to public if date is not available yet.
  **10th of May**
  4. If it's an update to existing version,  please select SDKs of specific language and CLIs that require refresh after swagger is published.
  **Not applicable**
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No, no need to refresh for updates in this PR

### Contribution checklist:
- [ ] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [ ] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist - (Not applicable)
- [ ] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  -  [ ] Ensure to copy the existing version into new directory structure for first commit (including refactoring) and then push new changes including version updates in separate commits. This is required to review the changes efficiently.
  - Adding a new service

- [ ] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist - (Not applicable)
If there are following updates in the PR, ensure to request an approval from Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

RGKarthik committed:
New Swagger Spec File

RGKarthik committed:
New Swagger Example Spec File

RGKarthik committed:
New Readme Config File

RGKarthik commented:
/azp run unifiedPipeline

RGKarthik commented:
/azp run unifiedPipeline

jhendrixMSFT commented on specification/agfood/data-plane/Microsoft.AgFoodPlatform/preview/2021-03-31-preview/agfood.json:
```diff
@@ -0,0 +1,132 @@
+        "x-ms-examples": {
+          "BatchAccountDelete": { "$ref": "./examples/OperationGroupGet.json" }
+        },
+        "description": "This is a sample get operation, please see guidelines in azure-rest-api-specs repository for more info",
```
Can you please clarify the intent of this PR?  This looks like some kind of sample content.

RGKarthik replied:
Joel,
Sending details in a mail. We are onboarding an new service onto Azure. Have updated the PR. Request your help to review the same.

RGKarthik committed:
AgFoodPlatform: Data plane swagger + examples (#14045)

* Uploading swagger and sample example

* Model validation fix attempt

* Fixing example

* Removing extra references to examples in swagger

* Making swagger pretty

* Spell check fix

* Added links to examples

* Added references to examples

* Fixing spell check error

* Fixed 2026 error

* Added dummy examples

* Fixing avocado errors

* Seasonal Field examples added

* FIx for seasonal field examples

* Adding examples for all objects

* Made examples prettier

* Updated request body in examples

* Debug effort v1

* Fixing issue in OAUthProvider example

* Time: removing millis

* Fixing date formats

* Fixing response codes

* Response code fix2

* Fixing object reference errors

* Fixing pipeline

* Fixed examples related errors

* Fixing more errors

* Down to 6 errors

* Fixing errors

* Fixing errors

* Fixing errors

* Fixin geometry related errors

* Fixing boundary errros

* Byte to binary

* Fixing download scenes error

* Trying fix for multipart-form

* Fixing duration in seconds

Co-authored-by: Karthik RG <RGKarthik@users.noreply.github.com>
Co-authored-by: Agnivesh Adhikari <agnivesh.adhikari@gmail.com>

jhendrixMSFT commented on specification/agfood/data-plane/Microsoft.AgFoodPlatform/preview/2021-03-31-preview/agfood.json:
```diff
@@ -0,0 +1,11996 @@
+              "$ref": "#/definitions/ApplicationData"
+            }
+          },
+          "404": {
```
Status codes listed under `responses` are considered success, so no error/exception will be propagated from the API call which I suspect isn't what you want.  If you want to keep this here for documentation purposes and have it be considered an error you must apply the [x-ms-error-response](https://github.com/Azure/autorest/tree/master/docs/extensions#x-ms-error-response) attribute.

RGKarthik replied:
Hi,
We do not want to propagate exceptions for 404. We use it a way to query if something exists or not.

jhendrixMSFT replied:
This authoring doesn't achieve that.  The 404 will be considered a success and the API will return an empty ApplicationData.

RGKarthik replied:
Fixed this.

jhendrixMSFT commented on specification/agfood/data-plane/Microsoft.AgFoodPlatform/preview/2021-03-31-preview/agfood.json:
```diff
@@ -0,0 +1,11996 @@
+          }
+        ],
+        "responses": {
+          "202": {
```
Per our conversation, annotate LROs with [x-ms-long-running-operation](https://github.com/Azure/autorest/tree/master/docs/extensions#x-ms-long-running-operation).

RGKarthik replied:
Fixed.

agniveshadhikari committed:
Farmbeats Swagger (#14203)

* big bang

* shapefile and rasterize are words

* adding back readme

* another copy pasta from v2

agniveshadhikari committed:
FarmBeats adding examples for rasterize (#14216)

* Adding examples for rasterize jobs

* Marking rasterize job as LRO

lmazuel committed:
Merge branch 'master' into dev-agfood-Microsoft.AgFoodPlatform-2021-03-31-preview

lmazuel committed:
LongRunningResponseStatusCode suppression

agniveshadhikari commented on custom-words.txt:
```diff
@@ -2158,4 +2158,16 @@ zipdeploy
-zset
\ No newline at end of file
+zset
+classificationrules
```
Do we want to keep this file sorted alphabetically?

jhendrixMSFT replied:
Yes please

agniveshadhikari reviewed (approved):
LGTM apart for the comment on custom words.

ellismg committed:
[FarmBeats] Use `x-ms-parameterized-host` (#14246)

Use the `x-ms-parameterized-host` pattern, which our code generators
understand and generate better code for.
